src/main/java/software/amazon/cloudwatchlogs/emf/logger/MetricsLogger.java

```diff
@@ -92,13 +94,14 @@ public void flush() {
             configureContextForEnvironment(context, environment);
             sink.accept(context);
             context = context.createCopyWithContext(flushPreserveDimensions);
+            Validator.metricNameAndResolutionMap.clear();
```

Comment:
We can avoid clearing this metricNameAndResolutionMap hashmap values in the validator by moving this logic to MetricsContext class.  

For each instance of MetricLogger we are creating new instance of MetricsContext class , so we can have this temp map created in the MetricContext class and validate the duplicate checks in the MetricContext class or pushing the assigned map object to validateMetric() method into Validator class. 

`public static void validateMetric(
            String name, double value, Unit unit, StorageResolution storageResolution,Map<String, String> metricNameAndResolutionMap)`

By doing this we don't have to maintain the state of map in the static class and avoid clearing the map content in each flush call.

Reply:
Thanks for the suggestion. Updated the code. Please take a look now.